<launch>

  
  <arg name="first_robot"  default="robot_0"/>
  <arg name="second_robot" default="robot_1"/>
  <arg name="third_robot"  default="robot_2"/>    
  <arg name="fourth_robot"  default="robot_3"/>
  <arg name="fifth_robot"  default="robot_4"/>
  <arg name="sixth_robot"  default="robot_5"/>
  <arg name="seventh_robot"  default="robot_6"/>
  <arg name="eighth_robot"  default="robot_7"/>

  <group ns="/robot_0/map_merge">
    <param name="init_pose_x" value="-4"/>
    <param name="init_pose_y" value="0"/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value="0"/>
  </group>
  <group ns="/robot_1/map_merge">
    <param name="init_pose_x" value="-3"/>
    <param name="init_pose_y" value="0 "/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value="0"/>
  </group>
  
  <group ns="/robot_2/map_merge">
    <param name="init_pose_x" value=" -2"/>
    <param name="init_pose_y" value="0"/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value=" 0"/>
  </group>

  <group ns="/robot_3/map_merge">
    <param name="init_pose_x" value=" -1"/>
    <param name="init_pose_y" value="0"/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value=" 0"/>
  </group>

  <group ns="/robot_4/map_merge">
    <param name="init_pose_x" value="  0"/>
    <param name="init_pose_y" value=" 0"/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value=" 0"/>
  </group>

  <group ns="/robot_5/map_merge">
    <param name="init_pose_x" value=" 1"/>
    <param name="init_pose_y" value=" 0"/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value=" 0"/>
  </group>

<!--   <group ns="/robot_6/map_merge">
    <param name="init_pose_x" value=" 2"/>
    <param name="init_pose_y" value=" 0"/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value=" 0"/>
  </group>
  <group ns="/robot_7/map_merge">
    <param name="init_pose_x" value=" 3"/>
    <param name="init_pose_y" value=" 0 "/>
    <param name="init_pose_z" value=" 0.0"/>
    <param name="init_pose_yaw"   value="0"/>
  </group> -->



    <node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge" output="screen">
      <param name="robot_map_topic" value="map"/>
      <param name="robot_namespace" value=""/>
      <param name="merged_map_topic" value="merged_map"/>
      <param name="world_frame" value="/map"/>
      <param name="known_init_poses" value="true"/>
      <param name="merging_rate" value="15"/>
      <param name="discovery_rate" value="0.05"/>
      <param name="estimation_rate" value="0.5"/>
      <param name="estimation_confidence" value="1.0"/>
    </node>

  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg first_robot)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg first_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg second_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg second_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg third_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg third_robot)/map 200"/>
    <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg fourth_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg fourth_robot)/map 200"/>  
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg fifth_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg fifth_robot)/map 200"/> 
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg sixth_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg sixth_robot)/map 200"/>
<!--   <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg seventh_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg seventh_robot)/map 200"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg eighth_robot)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg eighth_robot)/map 200"/> -->
  
  <node pkg="random_walk" name="coverage_percentage" type="map_compare.py" output="screen"></node> 

</launch>

 <!-- http://wiki.ros.org/multirobot_map_merge -->